### DISCUSSION TITLE
Stuck on "get" error

### DISCUSSION THREAD

**Discussion by rbgfloripa** [sindresorhus/ky#398] [General]
Hi all,

Can you guys help me , please ???

I have a React app and am using KY to fetch data from a Mulesoft integration.

My fetcher:

```javascript
export const fetcher = ky.create({
  prefixUrl: config.api.url,
  timeout: 5000,
  retry: false,
  mode: 'no-cors',
  headers: {
    'Content-Type': 'application/json',
  },
});
```
My fetch method: 

```javascript
fetcher.get('customers?personId=123454346&startDate=2021&endDate=2021')
  .then((res) => console.log(res))  // -> never call this console.log
  .catch((err) => console.log(err));  // -> always call this

```
I also tried with json()

```javascript
fetcher.get('customers?personId=123454346&startDate=2021&endDate=2021').json()
  .then((res) => console.log(res))  // -> never call this console.log
  .catch((err) => console.log(err));  // -> always call this

```
But the error is the same.

In the "network" tab of the browser, the request was **200** and I can see the response body.
However, console.log on "then" is never executed.
And the catch method is always triggered.

```
HTTPError: Request failed with status code 0
    HTTPError HTTPError.ts:15
    fn Ky.ts:42
    _retry Ky.ts:223
    create Ky.ts:63
    method index.ts:16
    fetchCustomers customers-model.js:68   // --> this line is fetcher.get(...)
```

PS: Sorry for my bad english  :-(

TIA

Rafael

**Comment by sholladay** [comment]
I don't know why the network tab would show you something different. I would recommend enabling "Preserve log" just so you can check that there's nothing super weird happening like a failing response and then a page refresh, followed by a successful request. I know that still wouldn't exactly match what you are describing, but it's worth trying anyway.

Your error message is coming from the following line in Ky:
https://github.com/sindresorhus/ky/blob/c315cc36e8989ef5bc7fe17111b87f6a8e46591f/source/core/Ky.ts#L42

If you're up for it, you could add a debugger breakpoint immediately before that line so that you can explore what the current state of the request, response, and the Ky instance is at the time that it fails. Or you could fork Ky and add some console logs around there. I'd also be happy to take a deeper look myself if you can provide a reproducible example.

**Comment by Vincent-Carrier** [comment]
Encountered the same problem today. Found the answer [here](https://stackoverflow.com/questions/45696999/fetch-unexpected-end-of-input
). Basically this is a CORS issue, unrelated to ky itself. But maybe ky should provide a hint in the form of a better error message. Maybe something along the lines of `Received opaque response. Verify server for CORS headers`?

**Reply by sholladay** [reply]
It's not a bad idea. In the case of an unknown status or code 0, we could link to our docs with some information about likely causes, including CORS.
